Game/MyProject/Game/SceneManagement/RoadNode.hpp

```diff
@@ -0,0 +1,19 @@
+
+#pragma once
+
+#include <IvMatrix33.h>
+#include <IvMath.h>
+#include <IvImage.h>
+#include "SceneNode.hpp"
+
+class RoadNode : std::enable_shared_from_this<RoadNode>, public SceneNode {
+
+public:
+    RoadNode(const std::string& name, std::vector<std::pair<IvVector3,IvVector3>> rMarginPoints, std::shared_ptr<SceneNode> graphRoot);
+    
+private:
+    std::shared_ptr<SceneNode> root;
+    std::vector<std::pair<IvVector3,IvVector3>> roadPoints;
```

Comment:
It seems that you don't need to keep nodes as members since you only build the mesh once in the constructor and then never use the points again.

Reply:
Maybe keep it for serialization later on..